pkg/storage/gcp/gcp.go

```diff
@@ -10,9 +10,16 @@ import (
 	"google.golang.org/api/option"
 )
 
+// Bucket defines the subset of functionality provided by the
+// cloud.google.com/go/storage.BucketHandle required by this package
+type Bucket interface {
+	Object(name string) *storage.ObjectHandle
+	Objects(ctx context.Context, q *storage.Query) *storage.ObjectIterator
+}
+
```

Comment:
Wouldn't it be easier to do something like this?
```
type Bucket interface {
	GetWriter(ctx context.Context, fileName string) *storage.Writer
	GetItems(ctx context.Context, module string) []string
}

```
and then something similar to:
```
func (b *gcpClient) GetWriter(ctx context.Context, fileName string) *storage.Writer {
  return  b.bkt.Object(fileName).NewWriter(ctx)
}
```
or some fake.
Otherwise you have to create all the intermediary objects like the ObjectHandle etc.

Reply:
Much easier :grin: I like it, I was dreading the next step with my impl.